Static content

blisteringherb opened the pull request:
Creating the static content content type.
- Includes the dosomething_static_content module
- Adding Entity Connect module to support adding or editing an image when referencing from a static content node
- Including the static content node template in the module
- Includes imagecache preset for gallery images

Fixes #494

Commit:
Adding the static content content type module

* Also adding the entity connect module to support creating a new
  entity when referencing it.

Commit:
Adding field collection module

Commit:
Fixing field collection merge conflict

Commit:
Adding additional fields and vars to static content type

* Including imagecache preset for galleries
* Overriding default location for static content temmplate file

Commit:
Fixing merge conflict

Commit:
Adding themed images and some markup

* Using image functions to add themed images to template
* Printing out available variables in the template

Commit:
Removing unnecessary $base_url

aaronschachter commented on drupal-org.make:
```diff
@@ -25,6 +25,10 @@ projects[diff][subdir] = "contrib"
 
+; Entity Connect
+projects[entity_connect] = "1.0-rc1"
+projects[entity][subdir] = "contrib"
```
should be projects[entity_connect]

blisteringherb replied:
I'm not sure what needs to be changed here.

aaronschachter commented on modules/dosomething/dosomething_static_content/dosomething_static_content.info:
```diff
@@ -0,0 +1,46 @@
+core = 7.x
+package = Features
+dependencies[] = ctools
+dependencies[] = dosomething_campaign
```
Why is this a dependency?

blisteringherb replied:
Not sure, removing it.

aaronschachter commented on modules/dosomething/dosomething_static_content/dosomething_static_content.module:
```diff
@@ -0,0 +1,69 @@
+/**
+ * Implements hook_theme_registry_alter()
+**/
+function dosomething_static_content_theme_registry_alter(&$theme_registry) {
```
Let's move this into a new general module -- dosomething_fuckit

angaither replied:
Perfect name.

aaronschachter commented on modules/dosomething/dosomething_static_content/dosomething_static_content.module:
```diff
@@ -0,0 +1,69 @@
+
+      $gallery_item_count = count($collection_item['#items']);
+      $vars['galleries'][$i] = array();
+      //$vars['galleries'][$i]['gallery_title'] = 
```
Stray comment?

aaronschachter commented on modules/dosomething/dosomething_static_content/node--static_content.tpl.php:
```diff
@@ -0,0 +1,37 @@
+  <?php endforeach; ?>
+
+  
+  <h2><?php print $additional_text_title; ?></h2>
```
Let's make sure to wrap all of these non-required variables in isset()

aaronschachter commented on modules/dosomething/dosomething_static_content/dosomething_static_content.module:
```diff
@@ -0,0 +1,69 @@
+      $vars['galleries'][$i] = array();
+      //$vars['galleries'][$i]['gallery_title'] = 
+      for ($a=0; $a<$gallery_item_count;$a++) {
+        $gallery_item = reset($collection_item[$a]['entity']['field_collection_item']);
```
This might be a dumb question but why do you need this reset() ?

aaronschachter commented:
Also need to update dosomething.info to include new dependencies:
- entityconnect
- dosomething_static_content

Commit:
Removing unnecessary dependency

Commit:
Creating a helper collection module for general DS helper functions

* Adding centralize theme registry alter to allow ds modules to register
  their paths for storing their respective template files.
* Adding strongarm variable for storing which DS modules should have
  their paths added to the theme registry.

Commit:
Checking for vars in tpl

Commit:
Fixing name of entity connect module

aaronschachter commented on drupal-org.make:
```diff
@@ -25,6 +25,10 @@ projects[diff][subdir] = "contrib"
 
+; Entity Connect
+projects[entityconnect] = "1.0-rc1"
+projects[entity][subdir] = "contrib"
```
I saw the commit where this was fixed, but why isn't it showing here?

aaronschachter replied:
Should be: `projects[entityconnect][subdir] = "contrib"`

aaronschachter commented:
dosomething.info will also need to be updated with any additional modules (dosomething_helpers)

Commit:
Adding entity connect to dosomething.info

Commit:
Fixing entityconnect make file typo

Commit:
Adding new modules to ds.info for makefile

Commit:
Merge branch 'dev' into static_content

angaither commented on modules/dosomething/dosomething_static_content/dosomething_static_content.module:
```diff
@@ -0,0 +1,41 @@
+function dosomething_static_content_preprocess_node(&$vars) {
+  if ($vars['type'] == 'static_content') {
+    $content = $vars['content'];
+    $vars['hero_image'] = dosomething_image_get_image($vars['field_hero_image'][0]['entity']->nid, 'horiz', 'hero_image');
```
these function names changed to `dosomething_image_get_themed_image`

Commit:
Updating image function name

angaither commented:
:shipit:
